test/shared_utils/reference_hashing.py

```diff
@@ -0,0 +1,190 @@
+"""A module containing common functionality used by multiple regression tests.
+
+These functions help build reference files that are scaled down from the raw
+expected test output. Instead of comparing netCDF4 files, a JSON file is
+produced that contains a hash of each group and variable. This hash includes:
+
+* Metadata attributes, excluding those with timestamps that vary based on when
+  the tests are executed.
+* Dimensions
+* (For variables) the values and shape of the data array.
+
+Main functions for external usage:
+
+* `create_xarray_reference_file` - for writing JSON structure of group and
+  variable hashes.
+* `get_hashes_from_xarray_input` - for iterating through groups and variables
+  in a file and generating a hash for each of them. This can be used both to
+  generate the reference file and to also convert request output during test
+  execution for comparison to that reference file.
+
+"""
+
+import json
+from hashlib import sha256
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+
+
+def create_xarray_reference_file(
+    input_file_path: str,
+    reference_file_path: str,
+    xarray_kwargs: dict = {'decode_timedelta': False},
```

Comment:
maybe we should turn off all xarray decoding? in case a version change of xarray blows up all of the regressions for who knows why.